File: timeboxing_back/Types/Day.cs
namespace timeboxing_back.Types
{
    public class Day
    {
        public DateOnly CurrentDay { get; init; }

        public DailyJobs DailyJobs { get; set; } = new DailyJobs();
        public JobsList JobsList { get; set; } = new JobsList();
        public Schedule Schedule { get; set; } = new Schedule();
        public Insights Insights { get; set; } = new Insights();

        public Day()
        {
            CurrentDay = DateOnly.FromDateTime(DateTime.Now);
        }

        public Day(DateOnly currentDay)
        {
            CurrentDay = currentDay;
        }
    }
}

File: timeboxing_front/src/types/Day.ts
import { DailyJobs } from "./DailyJobs";
import { Insights } from "./Insights";
import { JobsList } from "./JobsList";
import { Schedule } from "./Schedule";

export interface Day {
    DailyJobs: DailyJobs;
    JobsList: JobsList;
    Schedule: Schedule;
    Insights: Insights;
}